src/xtgeo/grid3d/_grid_import_xtgcpgeom.py

```diff
@@ -1,19 +1,68 @@
-# coding: utf-8
 """Private module, Grid Import private functions for xtgeo based formats."""
+from __future__ import annotations
 
 import json
 from struct import unpack
+from typing import TYPE_CHECKING, Literal, Tuple, TypedDict, Union, overload
 
 import h5py
 import numpy as np
+from typing_extensions import NotRequired, TypeAlias
```

Comment:
`typing_extensions` isn't in our dependency list, I'm guessing it's pulled in through something else. Should probably be added if we're using it